src/features/comments.ts

```diff
@@ -0,0 +1,114 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { createComment, deleteComment, getPostComments } from '../api/comments';
+import { Comment } from '../types/Comment';
+import { Post } from '../types/Post';
+
+type CommentsState = {
+  comments: Comment[];
+  loaded: boolean;
+  hasError: boolean;
+};
+
+const initialState: CommentsState = {
+  comments: [],
+  loaded: false,
+  hasError: false,
+};
+
+export const fetchComments = createAsyncThunk<Comment[], Post['id']>(
+  'comments/fetch',
+  async (postId, { rejectWithValue }) => {
+    try {
+      return await getPostComments(postId);
+    } catch (error) {
+      return rejectWithValue(
+        (error as Error).message || 'Failed to fetch comments',
+      );
+    }
+  },
+);
+
+export const addComment = createAsyncThunk<Comment, Omit<Comment, 'id'>>(
+  'comments/add',
+  async (comment, { rejectWithValue }) => {
+    try {
+      return await createComment(comment);
+    } catch (error) {
+      return rejectWithValue(
+        (error as Error).message || 'Failed to add comment',
+      );
+    }
+  },
+);
+
+export const removeComment = createAsyncThunk<number, Comment['id']>(
+  'comments/remove',
+  async (commentId, { rejectWithValue }) => {
+    try {
+      await deleteComment(commentId);
+
+      return commentId;
+    } catch (error) {
+      return rejectWithValue(
+        (error as Error).message || 'Failed to remove comment',
+      );
+    }
+  },
+);
+
+export const commentsSlice = createSlice({
+  name: 'comments',
+  initialState,
+  reducers: {},
+  extraReducers: builder => {
+    builder
+      .addCase(fetchComments.pending, state => ({
+        ...state,
+        loaded: false,
+        hasError: false,
+      }))
+      .addCase(fetchComments.fulfilled, (state, action) => ({
+        ...state,
+        comments: action.payload,
+        loaded: true,
+        hasError: false,
+      }))
+      .addCase(fetchComments.rejected, state => ({
+        ...state,
+        hasError: true,
+        loaded: true,
+        comments: [],
+      }));
```

Comment:
It seems like you are resetting the `comments` array to an empty array when fetching comments is rejected. This could lead to data loss in the UI, as the previously fetched comments would disappear. It might be better to keep the existing comments and just show an error message to the user.